tests/smoke_tests/checkers.py

```diff
@@ -0,0 +1,206 @@
+                    return [loss_tuple[1] for loss_tuple in losses_list]
+            else:
+                keyword = f"{self.loss_type.value}:"
+                if keyword in log_line:
+                    after_keyword = log_line.split(keyword)[-1]
+                    loss_string = after_keyword[: after_keyword.find("\t")]
+                    return [eval(loss_string)]
+        except Exception:
+            # if any parsing error occurs, return empty
+            return []
+        return []
+
+    def to_dict(self) -> Dict[str, Union[str, float]]:
+        return {
+            "loss": self.loss,
+            "loss_type": self.loss_type.value,
+            "metric_type": self.metric_type.value if self.metric_type is not None else "",
+            "tolerance": self.tolerance,
+        }
+
+
+class AccuracyChecker(MetricChecker):
+    def __init__(
+        self,
+        accuracy: float,
+        metric_type: MetricType,
+        scope: MetricScope = MetricScope.PREDICTION,
+        tolerance: float = DEFAULT_TOLERANCE,
+    ):
+        self.metric_type = metric_type
+        self.accuracy = accuracy
+        self.scope = scope
+        self.tolerance = tolerance
+        self.found_accuracy = False
+
+    def should_check(self, log_line: str) -> bool:
+        if self.metric_type == MetricType.TRAINING:
+            has_metrics = "metrics_distributed_fit" in log_line
+        else:
+            has_metrics = "metrics_distributed" in log_line
+
+        has_metrics |= f"Client {self.metric_type.value} Metrics" in log_line
+
+        return has_metrics
+
+    def check(self, log_line: str) -> None:
+        accuracies = self.parse_accuracies(log_line)
+        has_accuracy = any([approx(accuracy, abs=self.tolerance) == self.accuracy for accuracy in accuracies])
+
+        if has_accuracy:
+            self.found_accuracy = True
+
+    def parse_accuracies(self, log_line: str) -> List[float]:
+        try:
+            multiple_accuracies = False
+            if self.metric_type == MetricType.TRAINING:
+                if "metrics_distributed_fit" in log_line:
+                    keyword = "metrics_distributed_fit"
+                    multiple_accuracies = True
+                else:
+                    keyword = f"Client {self.metric_type.value} Metrics"
+            else:
+                if "metrics_distributed" in log_line:
+                    keyword = "metrics_distributed"
+                    multiple_accuracies = True
+                else:
+                    keyword = f"Client {self.metric_type.value} Metrics"
+
+            if keyword not in log_line:
+                return []
+
+            if self.metric_type == MetricType.TRAINING:
+                accuracy_type_keyword = f"train - {self.scope.value} - accuracy"
+            else:
+                accuracy_type_keyword = f"val - {self.scope.value} - accuracy"
+
+            if multiple_accuracies:
+                accuracies_string = log_line.split(keyword)[-1]
```

Comment:
Maybe we can encapsulate this parsing functionality for multiple or single accuracies into functions themselves for readability?